codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2HeadersDecoder.java

```diff
@@ -152,4 +162,24 @@ protected final boolean validateHeaders() {
     protected Http2Headers newHeaders() {
         return new DefaultHttp2Headers(validateHeaders, (int) headerArraySizeAccumulator);
     }
+
+    protected void validate(Http2Headers headers) throws Http2Exception {
+        Http2Headers.PseudoHeaderType prevType = null;
```

Comment:
in addition to verifying the type is the same as the previous, shouldn't we also provide an "expected type" to satisfy the following requirements:

> Pseudo-header fields defined for requests MUST NOT appear in responses;
pseudo-header fields defined for responses MUST NOT appear in requests.

Reply:
I'm going through h2spec failing tests and trying to fix them. The following tests should cover that

```
            <excludeSpec>8.1.2.3 - Sends a HEADERS frame with empty ":path" pseudo-header field</excludeSpec>
            <excludeSpec>8.1.2.3 - Sends a HEADERS frame that omits ":method" pseudo-header field</excludeSpec>
            <excludeSpec>8.1.2.3 - Sends a HEADERS frame that omits ":scheme" pseudo-header field</excludeSpec>
            <excludeSpec>8.1.2.3 - Sends a HEADERS frame that omits ":path" pseudo-header field</excludeSpec>
```

Reply:
@madgnome @Scottmitch  +1 I think we can fix these step by step

Reply:
I brought it up because it is mentioned in the commit message. If the intention is not to fix this in the current PR, please remove this text from the commit message.

Reply:
We check that pseudo headers are not mixed (no request header with response header or vice versa ) so it does meet the requirement in a way